File: static/js/projects.js
document.getElementById('add-project-form').addEventListener('submit', function (event) {
  event.preventDefault();

  const formData = new FormData(event.target);
  fetch('/add_project', {
    method: 'POST',
    body: formData,
  })
    .then((response) => response.json())
    .then((data) => {
      if (data.success) {
        // Append the new project to the table
        const newRow = document.createElement('tr');
        newRow.innerHTML = `
          <td>${data.project.id}</td>
          <td>${data.project.name}</td>
          <td>${data.project.effort_estimate}</td>
          <td>${data.project.start_date}</td>
          <td>${data.project.due_date}</td>
          <td>
            <button type="button" class="btn btn-danger" onclick="deleteProject({{ project.id }})">Delete</button>
          </td>
        `;
        document.getElementById('projects-tbody').appendChild(newRow);

        // Clear the form
        event.target.reset();

        // Show the Toast
        const toastElement = document.getElementById('project-added-toast');
        const toast = new bootstrap.Toast(toastElement);
        toast.show();
      } else {
        // Handle errors (e.g., display an error message)
        alert('Error adding project: ' + data.error);
      }
    });
});

document.getElementById('projects-tbody').addEventListener('click', function (event) {
  if (event.target.classList.contains('btn-delete')) {
    deleteProject(event);
  }
});



function deleteProject(projectId) {
  fetch('/delete_project/' + projectId, {
    method: 'DELETE',
  })
    .then((response) => response.json())
    .then((data) => {
      if (data.success) {
        // Remove the member row from the table
        const row = document.querySelector(`[data-project-id="${projectId}"]`);
        row.remove();
      } else {
        // Handle errors (e.g., display an error message)
        alert('Error deleting project: ' + data.error);
      }
    });
}

File: routes/projects.py
from flask import render_template, request, jsonify, Blueprint

from models.Project import Project
from models.db import Session

projects_bp = Blueprint('projects', __name__)


@projects_bp.route('/projects', methods=['GET'])
def projects():
    with Session() as s:
        projects = s.query(Project).all()
    return render_template('projects.html', projects=projects)


@projects_bp.route('/add_project', methods=['POST'])
def add_project():
    project_name = request.form['name']
    project_effort = int(request.form['effort_estimate'])
    project_start_date = request.form['start_date'] or None
    project_due_date = request.form['due_date'] or None

    if not all([project_name, project_effort]):
        return jsonify(success=False), 400

    # Check if member with the same email already exists
    with Session() as s:
        p = Project.add(s, name=project_name,
                        effort_estimate=project_effort,
                        start_date=project_start_date,
                        due_date=project_due_date)
        data = p.json()
        return jsonify(success=True, project=data), 201


@projects_bp.route('/delete_project/<int:project_id>', methods=['DELETE'])
def delete_project(project_id):
    with Session() as s:
        try:
            Project.remove_by_id(s, project_id)
            return jsonify(success=True), 200
        except Exception as e:
            return jsonify(success=False, error="Project not found"), 404
